dashboard/app/models/census/census_hoc2017v3.rb

```diff
@@ -0,0 +1,44 @@
+# == Schema Information
+#
+# Table name: census_submissions
+#
+#  id                           :integer          not null, primary key
+#  type                         :string(255)      not null
+#  submitter_email_address      :string(255)
+#  submitter_name               :string(255)
+#  submitter_role               :string(255)
+#  school_year                  :integer          not null
+#  how_many_do_hoc              :string(255)
+#  how_many_after_school        :string(255)
+#  how_many_10_hours            :string(255)
+#  how_many_20_hours            :string(255)
+#  other_classes_under_20_hours :boolean
+#  topic_blocks                 :boolean
+#  topic_text                   :boolean
+#  topic_robots                 :boolean
+#  topic_internet               :boolean
+#  topic_security               :boolean
+#  topic_data                   :boolean
+#  topic_web_design             :boolean
+#  topic_game_design            :boolean
+#  topic_other                  :boolean
+#  topic_other_description      :string(255)
+#  topic_do_not_know            :boolean
+#  class_frequency              :string(255)
+#  tell_us_more                 :string(255)
+#  pledged                      :boolean
+#  created_at                   :datetime         not null
+#  updated_at                   :datetime         not null
+#
+# Indexes
+#
+#  index_census_submissions_on_school_year_and_id  (school_year,id)
+#
+
+# This class represents submissions from the Hour of Code signup page
+# after the school autocomplete dropdown was added. Census questions were
+# not modified so no new logic is needed. We just need a new class to get
+# a new type in the DB.
+#
+class Census::CensusHoc2017v3 < Census::CensusHoc2017v2
```

Comment:
I don't know a ton about Ruby or ActiveRecord, so I have more questions than comments.

This is a fascinating approach.  I like the benefit of permanently encoding in our source the set of validations associated with any particular record.  Storing these types with single-table inheritance there are obviously changes we _could_ make to the table that would be destructive to past records (like removing a column), but we should be able to just choose not to make those destructive changes.

A question that I couldn't answer with a quick search: How expensive is it to have a very deep inheritance chain in Ruby, and particularly in ActiveRecord?  I don't expect the five-ish layers introduced in this PR to be problematic, but if we add another layer to the inheritance chain for every change to the census form I wonder if we'll hit a depth with a real performance cost.  Of course, based on the names here it looks like we're bounding this to the 2017 census and would start over for future years, so we're not expecting enough changes to cause a problem.